contrib/scalar/scalar.c

```diff
@@ -315,6 +315,45 @@ static char *remote_default_branch(const char *url)
 	return NULL;
 }
 
+static void strbuf_parentdir(struct strbuf *buf)
+{
+	int len = buf->len;
+	while (len > 0 && !is_dir_sep(buf->buf[--len]))
+		; /* keep looking for parent directory */
+	strbuf_setlen(buf, len);
```

Comment:
This loop appears earlier in the series, perhaps unify on the single implementation?

Reply:
Makes sense. We will do that as part of incorporating https://github.com/microsoft/git/pull/402, I think.

Reply:
Actually, I realized that you were talking about the code in `set_recommended_config()`, and did as  you suggested.